frontend/src/components/NavBar.js

```diff
@@ -107,10 +111,158 @@ class NavBar extends React.Component<{
+            {description && (
+                <div
+                    dangerouslySetInnerHTML={{ __html: description }}
+                    className='description'
+                ></div>
+            )}
+        </div>,
+    ];
+
+    handleSearchChange = (e, { value }) => {
+        this.setState({ isLoading: true, value });
+
+        setTimeout(() => {
+            if (this.state.value.length < 1) return this.setState(this.initialState);
+            const keywords = this.state.value.split(' ');
+            const regexKeywords = keywords.join('|');
+            const re = new RegExp(regexKeywords, 'gi');
+
+            const url = '/rest/interpret/wsearch';
+
+            $.ajax({
+                url: url,
+                dataType: 'json',
+                type: 'POST',
+                cache: false,
+                data: JSON.stringify({ keywords: keywords }),
+                contentType: 'application/json; charset=utf-8',
+                success: (data) => {
+                    /*
+                    Response body:
+                    ```
+                    {
+                        "response": [
+                            {id: 6,
+                            uuid: "0x5505f540936f4d0d919f3186141192b0",
+                            name: "codalab-a",
+                            title: "CodaLab Dashboard",
+                            frozen: null,
+                            owner_id: "0"
+                            owner_name: "codalab"
+                            group_permissions: {
+                                id: 8,
+                                group_uuid: "0x41e95d8592de417cbb726085d6986137",
+                                group_name: "public",
+                                permission: 1}
+                            }
+                            ...
+                        ]
+                    }
+                    ```
+
+                    turn the above response into the following dict
+                    ```
+                    {
+                        "name": "codalab",
+                        "results": [
+                            {
+                                "title": "Brakus Group",
+                                "description": "Cloned interactive Graphic Interface",
+                            },
+                        ]
+                    }
+                    ```
+                    */
+                    let filteredResults = {};
+                    for (let item of data.response) {
+                        // use DOMPurify to get rid of the XSS security risk
+                        item.description = DOMPurify.sanitize(
+                            item.name.replace(re, "<span id='highlight'>$&</span>"),
+                        );
+                        item.title = DOMPurify.sanitize(
+                            (item.title || '').replace(re, "<span id='highlight'>$&</span>"),
+                        );
```

Comment:
@candicegjing

Reply:
Yes. I just pulled from master and it works now.